ui/raidboss/data/05-shb/raid/e6s.js

```diff
@@ -7,6 +7,100 @@
   },
   timelineFile: 'e6s.txt',
   triggers: [
+    {
+      id: 'E6S Strike Spark',
+      regex: Regexes.startsUsing({ source: ['Ifrit', 'Raktapaksa'], id: '4BD3', capture: false }),
+      regexDe: Regexes.startsUsing({ source: ['Ifrit', 'Raktapaksa'], id: '4BD3', capture: false }),
+      regexFr: Regexes.startsUsing({ source: ['Ifrit', 'Raktapaksa'], id: '4BD3', capture: false }),
+      regexJa: Regexes.startsUsing({ source: ['イフリート', 'ラクタパクシャ'], id: '4BD3', capture: false }),
+      promise: function(data) {
+        let p = new Promise(async (res) => {
+          // helper function to delay the promise execution for the given time
+          const sleep = m => new Promise(r => setTimeout(r, m));
+
+          await sleep(10000);
+
+          let combatantNames = null;
+
+          // select the 4 most recent Ifrit or Raktapaksa's depending on phase
+          if (data.phase = 'ifrit') {
+              combatantNames = { names: ['Ifrit', 'イフリート', '伊弗利特', '이프리트'] };
+          } else {
+              // @TODO: Need the CN/KR names for Raktapaksa
+              combatantNames = { names: ['Raktapaksa', 'ラクタパクシャ'] };
+          }
+            
+          let combatantData = await window.callOverlayHandler({
+            call: 'getCombatants',
+            data: combatantNames
+          });
+
+          if (combatantData !== null &&
+            combatantData.combatants &&
+            combatantData.combatants.length) {
+              // helper function in order to check which one of the comparison,
+              // has the highest numeric value of the HEX ID
+              const idComparer = (id1, id2) => {
+                if (parseInt(id1, 16) > parseInt(id2, 16)) {
+                  return id1;
+                } else {
+                  return id2;
+                }
+              };
+
+              let currentHighestCombatant = null;
+
+              // we need to filter for the Ifrit with the highest HEX ID
+              // since that one is always the safe spot.
+              combatantData.combatants.forEach(combatant => {
```

Comment:
I know it's less efficient, but what about something like this for brevity:
`let currentHighestCombatant = combatantData.combatants.sort((a, b) => a.ID.localeCompare(b.ID)).pop();`

I think lexicographically, hex ids sort in order, so no need to parseInt them.